pca9685 error: 'delayMicroseconds' was not declared in this scope

### The problem

Sinds 2021.10. unable to compile any node with a pca9685 module.

src/esphome/components/pca9685/pca9685_output.cpp:72:24: error: 'delayMicroseconds' was not declared in this scope
   delayMicroseconds(500);



### Which version of ESPHome has the issue?

2021.10.X

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.10.

### What platform are you using?

ESP8266

### Board

nodemcu

### Component causing the issue

pca9685

### Example YAML snippet

```yaml
substitutions:
  node_name: garage_01

esphome:
  esp8266_restore_from_flash: true
  on_boot:
    - priority: 300
      then:
        - output.turn_off: 'pca9685_1_09'
        - output.turn_off: 'pca9685_1_10'
        - output.turn_off: 'pca9685_1_11'
        - output.turn_off: 'pca9685_1_12'




pca9685:
  - id: 'pca9685_1'
    frequency: 500
    address: 0x40
    
i2c:
  sda: D2
  scl: D1
  scan: True    

output:
  - platform: pca9685 
    pca9685_id: 'pca9685_1'
    id: pca9685_1_01
    channel: 0
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_02
    channel: 1
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_03
    channel: 2
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_04
    channel: 3
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_09
    channel: 8
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_10
    channel: 9
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_11
    channel: 10
    inverted: true
  - platform: pca9685
    pca9685_id: 'pca9685_1'
    id: pca9685_1_12
    channel: 11
    inverted: true


light:
  - platform: binary
    name: ga_lamp_boven
    output: pca9685_1_01
  - platform: binary
    name: ga_lamp_buiten
    output: pca9685_1_03

switch:
  - platform: output
    name: ga_compressor
    id: ga_compressor
    output: 'pca9685_1_02'
    internal: false
  - platform: output
    name: ga_socket
    id: ga_socket
    output: 'pca9685_1_04'
    internal: false
  - platform: output
    name: bu_sprinkler
    id: bu_sprinkler
    output: 'pca9685_1_09'
    internal: true
  - platform: output
    name: bu_sprinkler_a
    id: bu_sprinkler_a
    output: 'pca9685_1_10'
    internal: false
  - platform: output
    name: bu_sprinkler_v
    id: bu_sprinkler_v
    output: 'pca9685_1_11'
    internal: false
  - platform: output
    name: bu_sprinkler_m
    id: bu_sprinkler_m
    output: 'pca9685_1_12'
    internal: false
```


### Anything in the logs that might be useful for us?

```txt
INFO Reading configuration /config/esphome/garage_01.yaml...
WARNING 'garage_01': Using the '_' (underscore) character in the hostname is discouraged as it can cause problems with some DHCP and local name services. For more information, see https://esphome.io/guides/faq.html#why-shouldn-t-i-use-underscores-in-my-device-name
INFO Detected timezone 'Europe/Amsterdam'
INFO Detected timezone 'Europe/Amsterdam'
INFO Generating C++ source...
INFO Compiling app...
Processing garage_01 (board: nodemcuv2; framework: arduino; platform: platformio/espressif8266 @ 2.6.2)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES: 
 - framework-arduinoespressif8266 3.20704.0 (2.7.4) 
 - tool-esptool 1.413.0 (4.13) 
 - tool-esptoolpy 1.20800.0 (2.8.0) 
 - toolchain-xtensa 2.40802.200502 (4.8.2)
Dependency Graph
|-- <ESP8266WiFi> 1.0
|-- <noise-c> 0.1.3
|   |-- <libsodium> 1.10018.1
|-- <Wire> 1.0
|-- <ESP8266mDNS> 1.2
|   |-- <ESP8266WiFi> 1.0
Compiling /data/garage_01/.pioenvs/garage_01/src/esphome/components/pca9685/pca9685_output.cpp.o
Compiling /data/garage_01/.pioenvs/garage_01/src/esphome/components/sensor/sensor.cpp.o
src/esphome/components/pca9685/pca9685_output.cpp: In member function 'virtual void esphome::pca9685::PCA9685Output::setup()':
src/esphome/components/pca9685/pca9685_output.cpp:72:24: error: 'delayMicroseconds' was not declared in this scope
   delayMicroseconds(500);
                        ^
*** [/data/garage_01/.pioenvs/garage_01/src/esphome/components/pca9685/pca9685_output.cpp.o] Error 1
========================== [FAILED] Took 3.21 seconds ==========================
```


### Additional information

_No response_

--- a/esphome/components/pca9685/pca9685_output.cpp
+++ b/esphome/components/pca9685/pca9685_output.cpp
@@ -1,6 +1,7 @@
 #include "pca9685_output.h"
 #include "esphome/core/log.h"
 #include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
 
 namespace esphome {
 namespace pca9685 {
@@ -69,7 +70,7 @@ void PCA9685Output::setup() {
     this->mark_failed();
     return;
   }
-  delayMicroseconds(500);
+  esphome::delayMicroseconds(500);
 
   this->loop();
 }

--- a/esphome/components/pca9685/pca9685_output.cpp
+++ b/esphome/components/pca9685/pca9685_output.cpp
@@ -70,7 +70,7 @@ void PCA9685Output::setup() {
     this->mark_failed();
     return;
   }
-  esphome::delayMicroseconds(500);
+  delayMicroseconds(500);
 
   this->loop();
 }

